source/datetime_example.d

```diff
@@ -265,4 +265,30 @@ TimeOfDayは、DateTimeのうち、「何時何分何秒」の部分です。
     sw.reset();
 
     assert(sw.peek() <= t4);
+}
+
+/++
+# タイムゾーンの扱い
++/
+@safe unittest
+{
+    import core.time : hours;
+    import std.datetime.date : DateTime;
+    import std.datetime.systime: Clock, SysTime;
+    import std.datetime.timezone : LocalTime, SimpleTimeZone, UTC;
+
+    // LocalTimeはシステムのローカルタイムを返す。
+    auto tim1 = Clock.currTime();
+    assert(tim1.timezone is LocalTime());
+
+    // SysTimeにUTCを指定する。
+    auto tim2 = SysTime(DateTime(2019, 5, 1, 10, 0, 0), UTC());
+    assert(tim2.timezone is UTC());
+    assert(tim2.timezone.name == "UTC");
+    assert(tim2.toSimpleString() == "2019-May-01 10:00:00Z");
+
+    // 他のタイムゾーンに変換する。
+    auto JST = new immutable SimpleTimeZone(9.hours, "Asia/Tokyo");
```

Comment:
SimpleTimeZoneははzoneinfoを考慮しないので正しい使い方ではない旨を書くか、まるっと消してしまったほうがよいとおもいます。
https://dlang.org/library/std/datetime/timezone/simple_time_zone.html
> name and dstName are always the empty string since this time zone has no DST, and while it may be meant to represent a time zone which is in the TZ Database, obviously it's not likely to be following the exact rules of any of the time zones in the TZ Database, so it makes no sense to set it.

Reply:
OSに依存することなく書くことはやはり不可能なのか…？

Reply:
OS依存というよりも、なにがしかの方法で夏時間や閏秒のデータを保持する必要があります。
JavaやPHPはOSのzoneinfoを使わず自前でzoneinfoを持っていますが、zoneinfoを更新するためにはOSベンダのパッチとは別に処理系のパッチバージョンを更新しなければなりません。